test/format/smoke.sh

```diff
@@ -2,10 +2,15 @@
 
 set -e
 
+cache_max=$(free -m | grep -oP '\d+' | head -n 1)
+# Use special command to perform floating point number arithmetic and remove all decimals.
+cache_max=$(echo "0.2 * $cache_max" | bc | cut -d. -f1)
```

Comment:
Where did the 20% come from? It feels a little conservative, at least in the scenario where format is the only thing running.

Reply:
In this case it is possible that format runs in parallel with other tasks (e.g. make check)

Reply:
Right - I think we need to be smarter about this, including possible passing the limit in from whatever layer knows about the parallelism. (Evergreen?)

Reply:
Hmm I agree that we could probably have a better place for this but sadly i don't have a better idea. The reason behind that is Evergreen calls make check, which would go through all cmake configured Ctest labels and eventually run the test/format smoke.sh bash script. I was thinking that we could add the limit into cmake, but it would complicate lots of things (e.g. cmake language being difficult to work with, developers would find it difficult to find where the cache max is set).

Reply:
Yeah, I've come to agree. This context awareness is a bigger job, and the BFGs we've seen so far have all been for the standalone test, which this PR would address. So as discussed IRL, I'll close this thread out and we can cross this bridge when we come to it :-)